mmcv/transforms/formatting.py

```diff
@@ -0,0 +1,603 @@
+        if results.get('gt_bboxes', None) is not None:
+            bboxes = results['gt_bboxes'] * results['scale_factor']
+            if self.bbox_clip_border:
+                bboxes[:, 0::2] = np.clip(bboxes[:, 0::2], 0, results['width'])
+                bboxes[:, 1::2] = np.clip(bboxes[:, 1::2], 0,
+                                          results['height'])
+            results['gt_bboxes'] = bboxes
+
+    def _resize_keypoints(self, results: dict) -> None:
+        """Resize keypoints with ``results['scale_factor']``."""
+        if results.get('gt_keypoints', None) is not None:
+            keypoints = results['gt_keypoints'] * results['scale_factor'][:2]
+            if self.bbox_clip_border:
+                keypoints[:, 0::2] = np.clip(keypoints[:, 0::2], 0,
+                                             results['width'])
+                keypoints[:, 1::2] = np.clip(keypoints[:, 1::2], 0,
+                                             results['height'])
+            results['gt_keypoints'] = keypoints
+
+    def _resize_seg(self, results: dict) -> None:
+        """Resize semantic segmentation map with ``results['scale']``."""
+        if results.get('gt_semantic_seg', None) is not None:
+            if self.keep_ratio:
+                gt_seg = mmcv.imrescale(
+                    results['gt_semantic_seg'],
+                    results['scale'],
+                    interpolation=self.interpolation,
+                    backend=self.backend)
+            else:
+                gt_seg = mmcv.imresize(
+                    results['gt_semantic_seg'],
+                    results['scale'],
+                    interpolation=self.interpolation,
+                    backend=self.backend)
+            results['gt_semantic_seg'] = gt_seg
+
+    def transform(self, results: dict) -> dict:
+        """Transform function to resize images, bounding boxes, semantic
+        segmentation map.
+
+        Args:
+            results (dict): Result dict from loading pipeline.
+        Returns:
+            dict: Resized results, 'img', 'gt_bboxes', 'gt_semantic_seg',
+                'gt_keypoints', 'scale', 'scale_factor', 'height', 'width',
+                and 'keep_ratio' keys are updated in result dict.
+        """
+        if self.override:
+            self._random_scale(results)
+        else:
+            if 'scale' in results:
+                pass
+            elif 'scale' not in results and 'scale_factor' in results:
+                h, w = results['img'].shape[:2]
+                scale_factor = results['scale_factor']
+                results['scale'] = (int(scale_factor[0] * w + 0.5),
+                                    int(scale_factor[1] * h + 0.5))
+            else:
+                self._random_scale(results)
+
+        self._resize_img(results)
+        self._resize_bboxes(results)
+        self._resize_keypoints(results)
+        self._resize_seg(results)
```

Comment:
Can just use `Resize` instead of implementing `self._resize_xxx `.
Pass the parameters for instantiating `Resize`